src/Compile/CompilePanel.ts

```diff
@@ -132,13 +132,34 @@ export class CompilePanel {
   /**
    * Sets up an event listener to listen for messages passed from the webview context and
    * executes code based on the message that is recieved.
-   *
-   * @param webview A reference to the extension webview
-   * @param context A reference to the extension context
    */
   private _setWebviewMessageListener() {
-    /* NYI */
-    vscode.window.showInformationMessage('NYI');
+    const webview = this._panel.webview;
+    webview.onDidReceiveMessage((message: any) => {
+      const command = message.command;
+      const text = message.text;
+
+      switch (command) {
+        case 'request-compile':  // Called when [Compile] button was pressed
+          //
+          // TODO Run ONE compile command
+          //
+          vscode.window.showInformationMessage(text);
+          return;
+        case 'set-output-dir':  // Called to set output directory
+          const options = {canSelectMany: false, canSelectFiles: false, canSelectFolders: true};
+
+          vscode.window.showOpenDialog(options).then((val: vscode.Uri[]|undefined) => {
+            if (val !== undefined) {
+              this._panel.webview.postMessage({command: 'set-output-dir', data: val[0].fsPath});
+            }
```

Comment:
Q. When `val` is given as `undefined`?

Reply:
Actually, I am not sure. :-) 
At first, I thought that user could enter some non-existing path... and such could make `val === undefined`. However, it turned out that vscode UI (`vscode.window.showOpenDialog`) shows error UI when wrong path was entered, making user enter correct path. Therefore `val` should not be `undefined`.

However, I must check if `val !== undefined` in typescript because `val`'s type is `vscode.Uri[]|undefined`. So I added L153.